### PR TITLE
ADD: Scheduler cpu memory limits and retry

### PR DESCRIPTION
**Pull request by benleetownsend**

### CODE REVIEW

File: finetune/scheduler.py [hunk 1]
```diff
@@ -1,17 +1,54 @@
...
 def bytes_to_meg(x):
     return x / 1024 / 1024
 
+
+def scheduled(fn):
+    @functools.wraps(fn)
+    def scheduled_predict(self, model_file, x, *args, **kwargs):
+        model = self._rotate_in_model(model_file)
+        try:
+            preds = fn(self, model_file=model_file, x=x, *args, model=model, **kwargs)
```
Comment (madisonmay) [inline finetune/scheduler.py:24]: Trying to think if there are ever any "user error" style scenarios w/ bad inputs that might cause someone to purge everyone elses models.

File: finetune/scheduler.py [hunk 2]
```diff
@@ -23,89 +60,101 @@ def __init__(self, max_models=None, config=None, reserved=750000000):
...
 
-        if self.max_model_size is None or (in_use - self.previous_in_use) > self.max_model_size:
+        if (
+            self.max_model_size is None
+            or (in_use - self.previous_in_use) > self.max_model_size
+        ):
             self.max_model_size = in_use - self.previous_in_use
 
         self.previous_in_use = in_use
+        cpu_percent = psutil.virtual_memory().percent
```
Comment (madisonmay) [inline finetune/scheduler.py:78]: I'm unsure whether we should be tracking memory_percent() or virtual_memory().percent here

File: finetune/scheduler.py [hunk 3]
```diff
@@ -23,89 +60,101 @@ def __init__(self, max_models=None, config=None, reserved=750000000):
...
             del model.saver.variables
             del model.saver.fallback_
-            
+
         if self.session is None:
-            self.session = tf.Session() # delay this so that any options get applied from finetune.
-            self.gpu_memory_limit = self.session.run(tf.contrib.memory_stats.BytesLimit())
+            self.session = (
+                tf.Session()
+            )  # delay this so that any options get applied from finetune.
```
Comment (madisonmay) [inline finetune/scheduler.py]: Kind of confused why this is in parens

Comment (benleetownsend) [reply finetune/scheduler.py]: Black. I will manually put the comment on the next line.

**Review by madisonmay — APPROVED** [review]

**Pull request merged**